import { AxiosError } from 'axios';
import { GenericError, AxiosErrorContext } from '../types/Error';
import { HubSpotAuthError } from '../models/HubSpotAuthError';
export declare function isMissingScopeError(err: AxiosError<any>): boolean;
export declare function isGatingError(err: AxiosError<any>): boolean;
export declare function isApiUploadValidationError(err: AxiosError<any>): boolean;
export declare function isSpecifiedHubSpotAuthError(err: GenericError, { status, category, subCategory }: Partial<HubSpotAuthError>): boolean;
/**
 * @throws
 */
export declare function throwAxiosErrorWithContext(error: AxiosError<any>, context?: AxiosErrorContext): never;
/**
 * @throws
 */
export declare function throwApiError(error: AxiosError, context?: AxiosErrorContext): never;
/**
 * @throws
 */
export declare function throwApiUploadError(error: AxiosError, context?: AxiosErrorContext): never;
